engine/src/main/java/org/camunda/bpm/engine/HistoryService.java

```diff
@@ -107,21 +109,58 @@
+   *
+   * <p>The result of the query is empty when the user has no...
+   * <ul>
+   *   <li>{@link Permissions#READ_HISTORY} permission on {@link Resources#PROCESS_DEFINITION} OR
+   *   <li>{@link HistoricTaskPermissions#READ} permission on
+   *       {@link Resources#HISTORIC_TASK} ({@code enableHistoricInstancePermissions} in
+   *       {@link ProcessEngineConfigurationImpl} must be set to {@code true})
+   * */
   HistoricTaskInstanceQuery createHistoricTaskInstanceQuery();
 
-  /** Creates a new programmatic query to search for {@link HistoricDetail}s. */
+  /**
+   * <p>Creates a new programmatic query to search for {@link HistoricDetail}s.
+   *
+   * <p>The result of the query is empty when the user has no...
+   * <ul>
+   *   <li>{@link Permissions#READ_HISTORY} permission on
+   *       {@link Resources#PROCESS_DEFINITION} OR
+   *   <li>{@link HistoricTaskPermissions#READ} permission on
+   *       {@link Resources#HISTORIC_TASK} ({@code enableHistoricInstancePermissions} in
+   *       {@link ProcessEngineConfigurationImpl} must be set to {@code true}) OR
+   *   <li>{@link ProcessDefinitionPermissions#READ_HISTORY_VARIABLE} permission on
+   *       {@link Resources#PROCESS_DEFINITION}
+   *       ({@link ProcessEngineConfiguration#enforceSpecificVariablePermission} must be set to
+   *       {@code true}) OR
+   *   <li>{@link HistoricTaskPermissions#READ_VARIABLE} permission on
+   *       {@link Resources#HISTORIC_TASK} ({@code enforceSpecificVariablePermission} and
+   *       {@code enableHistoricInstancePermissions}
+   *       in {@link ProcessEngineConfigurationImpl} must be set to {@code true})
+   * */
   HistoricDetailQuery createHistoricDetailQuery();
 
   /**
-   * Creates a new programmatic query to search for {@link HistoricVariableInstance}s.
-   * <p>
-   * The result of the query is empty:
-   * <li>if the user has no {@link Permissions#READ_HISTORY} permission on {@link Resources#PROCESS_DEFINITION} or</li>
-   * <li>the user has no {@link ProcessDefinitionPermissions#READ_HISTORY_VARIABLE} permission on {@link Resources#PROCESS_DEFINITION}
-   * in case {@link ProcessEngineConfiguration#enforceSpecificVariablePermission} is enabled.</li>
-   * </p>
-   */
+   * <p>Creates a new programmatic query to search for {@link HistoricVariableInstance}s.
+   *
+   * <p>The result of the query is empty when the user has no...
+   * <ul>
+   *   <li>{@link Permissions#READ_HISTORY} permission on
+   *       {@link Resources#PROCESS_DEFINITION} OR
+   *   <li>{@link HistoricTaskPermissions#READ} permission on
+   *       {@link Resources#HISTORIC_TASK} ({@code enableHistoricInstancePermissions} in
+   *       {@link ProcessEngineConfigurationImpl} must be set to {@code true}) OR
+   *   <li>{@link ProcessDefinitionPermissions#READ_HISTORY_VARIABLE} permission on
+   *       {@link Resources#PROCESS_DEFINITION}
+   *       ({@link ProcessEngineConfiguration#enforceSpecificVariablePermission} must be set to
```

Comment:
to unify with the rest of the mentions
```suggestion
   *       ({@code enforceSpecificVariablePermission} in {@link ProcessEngineConfigurationImpl} must be set to
```